src/Parlot/Fluent/Parser.TryParse.cs

```diff
@@ -18,6 +28,30 @@ public T Parse(string text, ParseContext context = null)
             return default;
         }
 
+        private Parser<T> CheckCompiled(ParseContext context)
+        {
+            if (_compiledParser != null)
```

Comment:
`if (_compiledParser != null ||context.CompilationThreshold == 0) { return _compiledParser ?? this; }`, then later logic would only be run when it's active for sure, fast path ?